File: src/main/java/github/cweijan/ultimate/transaction/Transaction.java
package github.cweijan.ultimate.transaction;

import java.sql.Connection;
import java.sql.SQLException;

public interface Transaction{
    /**
     * Retrieve inner database connection
     * @return DataBase connection
     */
    Connection getConnection() throws SQLException;

    /**
     * Commit inner database connection.
     */
    void commit() throws SQLException;

    /**
     * Rollback inner database connection.
     */
    void rollback() throws SQLException;

    /**
     * Close inner database connection.
     */
    void close() throws SQLException;

}

File: src/main/java/github/cweijan/ultimate/transaction/Transaction.kt
package github.cweijan.ultimate.transaction

import github.cweijan.ultimate.db.config.DbConfig
import github.cweijan.ultimate.util.Log
import org.springframework.transaction.PlatformTransactionManager
import org.springframework.transaction.TransactionDefinition
import org.springframework.transaction.TransactionStatus

import java.sql.Connection
import java.sql.SQLException
import kotlin.math.log

class Transaction(var dbConfig: DbConfig) : PlatformTransactionManager {

    override fun getTransaction(definition: TransactionDefinition?): TransactionStatus {
        Log.info("gettransd")
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    fun begin() {
        Log.info("begin")
        dbConfig.openConnection().autoCommit = false
    }

    override fun commit(status: TransactionStatus) {
        Log.info("commit")
        dbConfig.openConnection().commit()
    }

    override fun rollback(status: TransactionStatus) {
        Log.info("rollback")
        dbConfig.openConnection().rollback()//事务提交
    }

}
